### PR TITLE
#9999: remove old maxpools

### PR DESCRIPTION
**Pull request by mywoodstock**
### Ticket
#9999 

### Problem description
Provide context for the problem.

### What's changed
Describe the approach used to solve the problem.
Summarize the changes made and its impact.

### Checklist
- [ ] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tests/ttnn/unit_tests/operations/test_maxpool2d.py [hunk 1]
```diff
@@ -212,3 +208,254 @@ def test_run_max_pool(
...
+
+    if in_n > 16 and in_c > 64 and dtype == ttnn.bfloat8_b and is_wormhole_b0():
+        pytest.skip("This case runs out of memory on Wormhole b0")
+
+    torch.manual_seed(0)
+    torch.set_printoptions(precision=3, sci_mode=False, linewidth=500, threshold=10000, edgeitems=32)
+
+    ## construct the tensor in NCHW shape
+    act = torch.randn(act_shape, dtype=torch.bfloat16)
+    # act = torch.zeros(act_shape, dtype=torch.bfloat16)
```
Comment (ayerofieiev-tt) [inline tests/ttnn/unit_tests/operations/test_maxpool2d.py:364]: remove?

Comment (mywoodstock) [reply tests/ttnn/unit_tests/operations/test_maxpool2d.py:364]: I like to keep these as they help in debugs :)

File: ttnn/ttnn/__init__.py [hunk 2]
```diff
@@ -304,7 +304,7 @@ def prelu(*args, **kwargs):  # Alias for leaky_relu. TODO(#8544): implement PReL
 )
 
 from ttnn.operations.conv2d import Conv2d, Conv2dConfig, get_conv_output_dim, get_conv_padded_input_shape_and_mem_config
-from ttnn.operations.pool import TTPyMaxPool, max_pool2d, max_pool2d_legacy, MaxPool2d, avg_pool2d
+from ttnn.operations.pool import avg_pool2d
```
Comment (ayerofieiev-tt) [inline ttnn/ttnn/__init__.py:307]: are you sure to remove `max_pool2d` from here too?

File: ttnn/cpp/ttnn/operations/pool/maxpool/max_pool2d_pybind.hpp [hunk 3]
```diff
@@ -11,5 +11,6 @@ namespace py = pybind11;
 namespace ttnn::operations::pool {
 
 void bind_max_pool2d_operation(py::module& module);
+void py_module(py::module& module);
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/ttnn/operations/pool/maxpool/max_pool2d_pybind.hpp:14]: why do you need this function?

File: ttnn/ttnn/__init__.py (hunk 2 above)
Comment (mywoodstock) [reply ttnn/ttnn/__init__.py:307]: yes, thats the old version

File: ttnn/cpp/ttnn/operations/pool/maxpool/max_pool2d_pybind.hpp (hunk 3 above)
Comment (mywoodstock) [reply ttnn/cpp/ttnn/operations/pool/maxpool/max_pool2d_pybind.hpp:14]: it calls the py bind function

Comment (mywoodstock) [reply ttnn/cpp/ttnn/operations/pool/maxpool/max_pool2d_pybind.hpp:14]: previously it was called in the old maxpool pybind, so just moving it here

File: tests/ttnn/unit_tests/operations/test_maxpool2d.py (hunk 1 above)
Comment (yan-zaretskiy) [reply tests/ttnn/unit_tests/operations/test_maxpool2d.py:364]: I'd encourage you to turn this into a testing util function then. Someone might delete this code in another PR as dead.

**Review by yan-zaretskiy — APPROVED** [review]

**Comment by sankarmanoj-tt** [comment]
As the old max pools are removed, I think it would be better to call the existing one as just `max_pool2d` instead of `max_pool2d_new`

**Review by pavlejosipovic — APPROVED** [review]

**Review by shwetankTT — APPROVED** [review]

**Comment by mywoodstock** [comment]
> As the old max pools are removed, I think it would be better to call the existing one as just `max_pool2d` instead of `max_pool2d_new`

yes will be updating that next.

**Review by esmalTT — APPROVED** [review]

**Pull request merged**